src/viam/sdk/resource/resource.hpp

```diff
@@ -28,6 +28,8 @@ class Resource {
     /// @brief Return the resource's name.
     virtual std::string name() const;
 
+    std::vector<Name> resource_names() const;
```

Comment:
Let's please add a doc comment for this while we are moving it.

Also, is `resource_names` the best name for this method? I'm actually a little unclear on what the meaning of the returned names is. It seems to be all the registered models that share the same API as this resource?

Reply:
Done! Though, it appears the only place this method is ever used is on a private method in `RobotService`. I wonder if it might make sense to just have this be a private method and make `RobotService` a friend class?